MIP/mip-27/README.md

```diff
@@ -0,0 +1,126 @@
+# MIP-27: Contract Pipeline
+
+- **Description**: Proposal to establish a structured pipeline for new contract deployments, aligning with the infrastructure outlined in MIP-23 while maintaining the existing Movement Token under MIP-18.
+- **Authors**: [Primata](mailto:primata@movementlabs.xyz)
+
+## Abstract
+
+MIP-27 proposes a Contract Pipeline that leverages a timelock-based infrastructure, ensuring security and decentralization for new contract deployments. While the Movement Token (MOVEToken) continues to operate under the provisions of MIP-18, future deployments should adhere to the structure set forth by MIP-23. The pipeline includes components such as a KMS multisig for transaction proposals, off-chain voting for stakers and token holders, and a timelock mechanism for executing transactions, providing transparency and decentralized control.
+
+## Motivation
+
+Currently, Movement Token operates under the governance outlined in [MIP-18](https://github.com/movementlabsxyz/MIP/blob/primata/mip-2/MIP/mip-18/README.md), and we may wish to maintain this setup for the time being. However, for future deployments and upgrades, we should transition to a more robust infrastructure as described in MIP-23. This approach provides decentralized control, security, and accountability through multisig setups, timelocks, and community voting mechanisms. By establishing a structured pipeline, we ensure that the development process remains transparent, auditable, and aligned with the community's interests.
+
+## Specification
+
+The Contract Pipeline includes the following components:
+
+### 1. **Timelock**  
+
+- **Role**: The timelock holds ownership of all upgradeable contracts.
+- **Purpose**: Ensures that contract upgrades are executed only after a specified delay, allowing time for review and potential cancellation by stakeholders.
+
+### 2. **KMS Multisig Setup**  
+
+- **Role**: Proposes transactions to the timelock.
+- **Components**:
+  - **1 KMS Key**: Proposes transactions based on accepted tags.
+  - **3 KMS Keys**: Accept transactions only if the transaction hash is contained in a GitHub release.
+  - **1 KMS Key**: Schedules the timelock transaction.
+
+### 3. **Off-Chain Voting Mechanism**  
+
+- **Role**: Allows stakers and token holders to participate in "heat-checking" decisions, where off-chain voting provides feedback before finalizing transaction proposals.
+- **Purpose**: Increases community involvement and allows for decentralized input into key governance decisions.
+
+### 4. **Engineering Crew**  
+
+- **Role**: The engineering team has the ability to push new contract versions for upgrades.
+- **Interaction**: The engineering team collaborates with the multisig setup to ensure proposed transactions align with the accepted development standards.
+
+### 5. **MCR Domain**  
+
+- **Role**: This domain is composed of network stakers who can execute timelocked transactions.
```

Comment:
Are you using "Domain" in the same sense as its usage in MCR?

Reply:
Yes